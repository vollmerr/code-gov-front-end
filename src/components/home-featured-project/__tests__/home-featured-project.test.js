import React from 'react';

import HomeFeaturedProject from 'components/home-featured-project'
import CustomLink from 'components/custom-link'

import { shallow } from 'enzyme';

describe('Testing <HomeFeaturedProject /> component', () => {
  const props = {
    index: 0,
    project: {
      alt: "test-alt",
      description: "test-description",
      image: "fa fa-comments",
      short_name: "test-short-name",
      links: [
        {
          name: "test-link-name",
          url: "https://test-link.com"
        }
      ]
    }
  }
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<HomeFeaturedProject {...props} />);
  });
  it('Testing if component renders', () => {
    expect(wrapper.exists()).toEqual(true);
  });
  it('Testing a link that does not start with "http"', () => {
    const props = {
      index: 0,
      project: {
        alt: "test-alt",
        description: "test-description",
        image: "test-image",
        short_name: "test-short-name",
        links: [
          {
            name: "test-link-name",
            url: "test-link"
          }
        ]
      }
    }
    wrapper = shallow(<HomeFeaturedProject {...props} />);
    expect(wrapper.exists()).toEqual(true);
  });
  it('should render correct', () => {
    expect(wrapper).toMatchSnapshot();
  })
})