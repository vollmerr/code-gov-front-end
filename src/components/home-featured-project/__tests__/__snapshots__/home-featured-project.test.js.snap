// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HomeFeaturedProject /> component should render correct 1`] = `
<div
  className="block featured-project"
>
  <div
    className="indented"
  >
    <hr />
    <div
      className="width-third"
    >
      <h2
        className="fp-short-name"
      >
        test-short-name
      </h2>
      <div
        align="center"
      >
        <i
          alt="test-alt"
          className="fa fa-comments"
        />
      </div>
    </div>
    <div
      className="width-two-thirds"
    >
      <p
        className="fp-description"
      >
        test-description
      </p>
      <span
        key="https://test-link.com"
      >
        <a
          href="https://test-link.com"
        >
          <button>
            test-link-name
          </button>
        </a>
      </span>
    </div>
  </div>
</div>
`;
